openedx/core/djangoapps/course_roles/tests/test_api.py

```diff
@@ -0,0 +1,135 @@
+"""
+Tests of the course_roles.helpers module
+"""
+import ddt
+from organizations.tests.factories import OrganizationFactory
+import pytest
+
+from opaque_keys.edx.keys import CourseKey
+
+from common.djangoapps.student.tests.factories import AnonymousUserFactory, UserFactory
+from openedx.core.djangoapps.content.course_overviews.models import CourseOverview
+from openedx.core.djangoapps.course_roles.api import (
+    get_all_user_permissions_for_a_course,
+)
+from openedx.core.djangoapps.course_roles.data import CourseRolesPermission
+from openedx.core.djangoapps.course_roles.models import (
+    Permission,
+    Role,
+    Service,
+    UserRole
+)
+from xmodule.modulestore.tests.django_utils import SharedModuleStoreTestCase
+from xmodule.modulestore.tests.factories import CourseFactory
+
+
+@ddt.ddt
+class GetAllUserPermissionsTestcase(SharedModuleStoreTestCase):
```

Comment:
A `SharedModuleStoreTestCase` only resets the ModuleStore information once for the test class, not for each test method. The `setUp` you're doing below this is run before each test case. This may trip folks up in the future since Modulestore entries will accumulate across tests. If you're only setting up the one `CourseFactory`, you can put that in this class's `setUpClass` method and leave the rest of the `setUp` as is.

If test performance matters, you can also streamline ORM data creation so that it only happens once, but you'd need to use a combination of `setUpTestData`, `setUpClass`, and the kind of weird `setUpClassAndTestData`–it's probably not worth it for you right now.

If you were creating different courses in the ModuleStore for each test, then you'd want to use `ModuleStoreTestCase`.